src/main/java/Task/TaskList.java

```diff
@@ -0,0 +1,124 @@
+package Task;
+
+import java.util.ArrayList;
+
+/**
+ * Manages a list of tasks, providing operations to manipulate and query tasks.
+ */
+public class TaskList {
+
+    public static final TaskList mainTaskList = new TaskList();
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Initializes a new empty list of tasks.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a task to the task list.
+     * @param task The task to add.
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     * @return The number of tasks.
+     */
+    public int getNumTasks() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Prints all tasks in the list with their indices.
+     */
+    public void printList() {
+        int index = 1;
+        System.out.println();
+        for (Task task : this.tasks) {
+            System.out.printf("     %d. %s\n", index, task);
+            index++;
+        }
+        System.out.println();
+    }
+
+    /**
+     * Finds and returns a new TaskList containing all tasks that include the specified term.
+     * @param term The term to search for within task descriptions.
+     * @return A TaskList of all tasks containing the term.
+     */
+    public TaskList tasksContainingTerm(String term) {
```

Comment:
generally, since this is a method, it should be a verb - `getTasksContainingTerm` to indicate that it is a method instead of a variable